src/main/java/net/minecraftforge/event/entity/player/CriticalHitEvent.java

```diff
@@ -0,0 +1,54 @@
+package net.minecraftforge.event.entity.player;
+
+import net.minecraftforge.fml.common.eventhandler.Event.HasResult;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+
+/**
+ * AttackEntityEvent is fired when a player is about to make a critical hit.<br>
+ * This event is fired whenever a player attacks an Entity in
+ * EntityPlayer#attackTargetEntityWithCurrentItem(Entity).<br>
+ * <br>
+ * {@link #livingTarget} contains the Entity that was damaged by the player. <br>
+ * <br>
+ * This event is not {@link Cancelable}.<br>
+ * If this event is canceled, the player does not make a critical hit (but he still will attack!).<br>
+ * <br>
+ * This event has a result. {@link HasResult}<br>
+ * DEFAULT: means the vanilla logic will determinate if this a critical hit. Else the vanilla logic is never used.<br>
```

Comment:
what does "Else the vanilla logic is never used"? even mean? Default just means "use vanilla logic". no further comment is needed